src/BatteryStateOfChargeService.h

```diff
@@ -17,6 +18,11 @@ class BatteryStateOfChargeService : public I_BatteryStateOfChargeService
 private:
     // Convention is to have an _ after private members
     const double initialStateOfChargePercent_;
+    QTime myTime;
+    QTime *prevTime;
+    double myVoltage;
+    double myCurrent;
```

Comment:
Using `my` is not very descriptive to indicate for the current `batteryData`, you should look for another prefix to indicate that :)